XamarinApp.Test/ViewModels/SignUpPageViewModelTest.cs

```diff
@@ -0,0 +1,82 @@
+using AutoFixture;
+using Moq;
+using Prism.Navigation;
+using Prism.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using XamarinApp.Models;
+using XamarinApp.Services;
+using XamarinApp.Services.Interfaces;
+using XamarinApp.ViewModels;
+using Xunit;
+
+namespace XamarinApp.Test.ViewModels
+{
+  public class SignUpPageViewModelTest
+  {
+    private readonly Mock<INavigationService> _navigationService;
+    private readonly Mock<IPageDialogService> _pageDialogService;
+    private readonly Fixture _fixture = new Fixture();
+    private readonly Mock<ISignUpUserService> _serviceMock; 
+    private readonly SignUpPageViewModel viewModel;
+
+    public SignUpPageViewModelTest()
+    {
+      _navigationService = new Mock<INavigationService>();
+      _pageDialogService = new Mock<IPageDialogService>();
+      _serviceMock = new Mock<ISignUpUserService>();
+      viewModel = new SignUpPageViewModel(_navigationService.Object, _pageDialogService.Object,_serviceMock.Object);
+    }
+
+    [Fact]
+    public void When_User_Click_SignIn_Navigate_to_LoginPage()
+    {
+      //Act
+      _navigationService.Setup(n => n.NavigateAsync("LoginPage")).ReturnsAsync(_fixture.Create<NavigationResult>());
+      viewModel.LoginCommand.Execute(new object());
+
+      //Assert
+      _navigationService.Verify(n => n.NavigateAsync("LoginPage"));
+    }
+
+    [Fact]
+    public void When_User_Enters_Valid_Email_and_Password_Navigate_to_LoginPage()
+    {
+      //Arrange
+      var user = _fixture.Create<UserModel>();
+
+      //Act
+      viewModel.UserName = user.UserName;
+      viewModel.PassWord = user.Password;
+      viewModel.ConfirmPassWord = user.Password;
+      _navigationService.Setup(n => n.NavigateAsync("LoginPage")).ReturnsAsync(_fixture.Create<NavigationResult>());
+      _serviceMock.Setup(n => n.SaveUser(It.Is<UserModel>(x => x.UserName == user.UserName && x.Password == user.Password))).ReturnsAsync(true);
+      viewModel.SignUpCommand.Execute(new object());
```

Comment:
you don't need to pass anything. It has a parameterless method also. check at other places as well.